drivers/spi/spi_xec_qmspi_v2.c

```diff
@@ -0,0 +1,746 @@
+ * Initialize SPI context.
+ * QMSPI will be fully configured and enabled when the transceive API
+ * is called.
+ */
+static int xec_qspi_init(const struct device *dev)
+{
+	const struct spi_xec_qspi_config *cfg = dev->config;
+	struct spi_xec_qspi_data * const qdata = dev->data;
+	struct qspi_regs * const regs = cfg->regs;
+	int ret = 0;
+
+	qdata->qstatus = 0;
+	qdata->np = cfg->width;
+
+	z_mchp_xec_pcr_periph_sleep(cfg->pcr_idx, cfg->pcr_pos, 0);
+
+	ret = pinctrl_apply_state(cfg->pcfg, PINCTRL_STATE_DEFAULT);
+	if (ret != 0) {
+		LOG_ERR("QSPI pinctrl setup failed (%d)", ret);
+		return ret;
+	}
+
+	xec_qspi_reset(regs);
+
+	spi_context_unlock_unconditionally(&qdata->ctx);
+
+	return 0;
+}
+
+static const struct spi_driver_api spi_xec_qspi_driver_api = {
+	.transceive = xec_qspi_transceive,
+	.release = xec_qspi_release,
+#ifdef CONFIG_SPI_ASYNC
+	.transceive_async = xec_qspi_transceive_async,
+#endif
+};
+
+#define XEC_QSPI_CS_TIMING_VAL(a, b, c, d) (((a) & 0xFu) \
+					    | (((b) & 0xFu) << 8) \
+					    | (((c) & 0xFu) << 16) \
+					    | (((d) & 0xFu) << 24))
+
+#define XEC_QSPI_TAPS_ADJ_VAL(a, b) (((a) & 0xffu) | (((b) & 0xffu) << 8))
+
+#define XEC_QSPI_CS_TIMING(i) XEC_QSPI_CS_TIMING_VAL(			\
+				DT_INST_PROP_OR(i, dcsckon, 6),		\
+				DT_INST_PROP_OR(i, dckcsoff, 4),	\
+				DT_INST_PROP_OR(i, dldh, 6),		\
+				DT_INST_PROP_OR(i, dcsda, 6))
+
+#define XEC_QSPI_TAPS_ADJ(i) XEC_QSPI_TAPS_ADJ_VAL(			\
+				DT_INST_PROP_OR(i, tctradj, 0),		\
+				DT_INST_PROP_OR(i, tsckadj, 0))
+
+#define XEC_QSPI_GIRQ(i)						\
+	MCHP_XEC_ECIA_GIRQ(DT_INST_PROP_BY_IDX(i, girqs, 0))
+
+#define XEC_QSPI_GIRQ_POS(i)						\
+	MCHP_XEC_ECIA_GIRQ_POS(DT_INST_PROP_BY_IDX(i, girqs, 0))
+
+#define XEC_QSPI_NVIC_AGGR(i)						\
+	MCHP_XEC_ECIA_NVIC_AGGR(DT_INST_PROP_BY_IDX(i, girqs, 0))
+
+#define XEC_QSPI_NVIC_DIRECT(i)					\
+	MCHP_XEC_ECIA_NVIC_DIRECT(DT_INST_PROP_BY_IDX(i, girqs, 0))
+
+/*
+ * The instance number, i is not related to block ID's rather the
+ * order the DT tools process all DT files in a build.
+ */
+#define XEC_QSPI_DEVICE(i)						\
+									\
+	PINCTRL_DT_INST_DEFINE(i);					\
+									\
+	static struct spi_xec_qspi_data xec_qspi_data_##i = {		\
+		SPI_CONTEXT_INIT_LOCK(xec_qspi_data_##i, ctx),		\
+		SPI_CONTEXT_INIT_SYNC(xec_qspi_data_##i, ctx),		\
+	};								\
+	static const struct spi_xec_qspi_config xec_qspi_config_##i = {	\
+		.regs = (struct qspi_regs *) DT_INST_REG_ADDR(i),	\
+		.cs1_freq = DT_INST_PROP_OR(i, cs1-freq, 0),		\
+		.cs_timing = XEC_QSPI_CS_TIMING(i),			\
+		.taps_adj = XEC_QSPI_TAPS_ADJ(i),			\
+		.girq = XEC_QSPI_GIRQ(i),				\
+		.girq_pos = XEC_QSPI_GIRQ_POS(i),			\
+		.girq_nvic_aggr = XEC_QSPI_NVIC_AGGR(i),		\
+		.girq_nvic_direct = XEC_QSPI_NVIC_DIRECT(i),		\
+		.irq_pri = DT_INST_IRQ(i, priority),			\
+		.pcr_idx = DT_INST_PROP_BY_IDX(i, pcrs, 0),		\
+		.pcr_pos = DT_INST_PROP_BY_IDX(i, pcrs, 1),		\
+		.port_sel = DT_INST_PROP_OR(i, port-sel, 0),		\
+		.chip_sel = DT_INST_PROP_OR(i, chip-select, 0),		\
```

Comment:
This is not correct either; what's going on?

Reply:
chip_sel is being used in the driver.

Reply:
updated accordingly